### ISSUE TITLE
Eslint config suggested improvement

### ISSUE THREAD

**Issue by devinrhode2** [TanStack/router#3045]
### Which project does this relate to?

Router

### Describe the bug

```js
'@typescript-eslint/only-throw-error': [
  'error',
  {
    allow: ['Redirect'],
    allowThrowingAny: false,
    allowThrowingUnknown: false,
  },
],
```

### Your Example Website or App

google.com

### Steps to Reproduce the Bug or Issue

write `throw redirect`

### Expected behavior

no eslint error when using strict @typescript-eslint config

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

**Comment by schiller-manuel** [comment]
should this be put into the docs?

**Comment by chrisbenincasa** [comment]
Thanks for this! Also added `notFound` to the list of allowed things to throw for the same reason.

**Comment by devinrhode2** [comment]
Could this go into the (new to me) `@tanstack/eslint-plugin-router` eslint plugin?

**Comment by SeanCassiere** [comment]
> Could this go into the (new to me) `@tanstack/eslint-plugin-router` eslint plugin?

I don't think we can (or should) be overriding another eslint plugin from ours.
